lib/puppet/indirector/catalog/compiler.rb

```diff
@@ -98,6 +146,16 @@ def compile(node, code_id)
       end
     end
 
+    if node.environment.static_catalogs? && options[:static_catalog] && config.is_a?(model)
+      str = "Inlined resource metadata into static catalog for #{node.name}"
+      str += " in environment #{node.environment}" if node.environment
+      benchmark(:notice, str) do
+        Puppet::Util::Profiler.profile(str, [:compiler, :static_inline, node.environment, node.name]) do
+          config = inline_metadata(config, node.environment, options[:checksum_type])
```

Comment:
`inline_metadata` only takes two arguments, looks like `node.environment` is extra.